oadr2/poll.py

```diff
@@ -223,106 +187,15 @@ def send_reply(self, payload, uri):
-    '''
-
-    def __init__(self, key, cert, ca_certs, ssl_version=None, ciphers=None):
-        '''
-        Intiailize the Client Authentication Handler
-        
-        key -- An encryption key
-        cert -- A Certificate
-        ca_certs -- CA Certificates
-        ssl_version -- What version of SSL are we using
-        ciphers -- What encryption method
-        '''
-
-        urllib.request.HTTPSHandler.__init__(self)
-        self.key = key
-        self.cert = cert
-        self.ca_certs = ca_certs
-        self.ssl_version = ssl_version
-        self.ciphers = ciphers
-
-    def https_open(self, req):
-        '''
-        Open a connection.
-        Rather than pass in a reference to a connection class, we pass in
-        a reference to a function which, for all intents and purposes,
-        will behave as a constructor
-
-        req -- A request
-        '''
-
-        return self.do_open(self.get_connection, req)
-
-    def get_connection(self, host, timeout=REQUEST_TIMEOUT):
-        '''
-        Gets a HTTPS connection.
-
-        host -- What host
-        timeout -- How long to wait.
-
-        Return HTTPSConnection object
-        '''
-
-        return HTTPSConnection(host,
-                               key_file=self.key,
-                               cert_file=self.cert,
-                               timeout=timeout,
-                               ciphers=self.ciphers,
-                               ca_certs=self.ca_certs,
-                               ssl_version=self.ssl_version)
-
-
-class HTTPSConnection(http.client.HTTPSConnection):
-    '''
-    Overridden to allow peer certificate validation, configuration
-    of SSL/ TLS version and cipher selection.  See:
-    http://hg.python.org/cpython/file/c1c45755397b/Lib/httplib.py#l1144
-    and `ssl.wrap_socket()`
-    '''
-
-    def __init__(self, host, **kwargs):
-        '''
-        Initilize the HTTPS conneciton
-
-        host -- See httplib.HTTPSConnection's documentation
-        kwargs -- See httplib.HTTPSConnection's documentation
-        '''
-
-        self.ciphers = kwargs.pop('ciphers', None)
-        self.ca_certs = kwargs.pop('ca_certs', None)
-        self.ssl_version = kwargs.pop('ssl_version', ssl.PROTOCOL_SSLv23)
-
-        http.client.HTTPSConnection.__init__(self, host, **kwargs)
-
-    def connect(self):
-        '''
-        Connect to the server.
-        '''
-
-        sock = socket.create_connection((self.host, self.port), self.timeout)
-
-        if self._tunnel_host:
-            self.sock = sock
-            self._tunnel()
-
-        if self.ca_certs and not os.path.isfile(self.ca_certs):
-            logging.warn("CA certs file does not exist: %s", self.ca_certs)
+        resp = requests.post(
+            uri,
+            cert=(
+                self.ven_client_cert_pem,
+                self.ven_client_cert_key
+            ) if self.ven_client_cert_pem and self.ven_client_cert_key else None,
```

Comment:
How about pulling it outside of the argument list?

Reply:
Changed